fury/actors/effect_manager.py

```diff
@@ -0,0 +1,431 @@
+        sigmas : np.ndarray (1, ) or (N, 1)
+            Array of sigmas to be used in the KDE calculations. Must be one or one for each point.
+        kernel : str, optional
+            Kernel to be used for the distribution calculation. The available options are:
+            * "cosine"
+            * "epanechnikov"
+            * "exponential"
+            * "gaussian"
+            * "linear"
+            * "tophat"
+
+        opacity : float, optional
+            Opacity of the actor.
+        colormap : str, optional.
+            Colormap matplotlib name for the KDE rendering. Default is "viridis".
+        custom_colormap : np.ndarray (N, 4), optional
+            Custom colormap for the KDE rendering. Default is none which means no 
+            custom colormap is desired. If passed, will overwrite matplotlib colormap 
+            chosen in the previous parameter.
+            
+        Returns
+        -------
+        textured_billboard : actor.Actor
+            KDE rendering actor."""
+        if not isinstance(sigmas, np.ndarray):
+            sigmas = np.array(sigmas)
+        if sigmas.shape[0] != 1 and sigmas.shape[0] != points.shape[0]:
+            raise IndexError("sigmas size must be one or points size.")
+        if np.min(sigmas) <= 0:
+            raise ValueError("sigmas can't have zero or negative values.")
+
+        varying_dec = """
+        varying float out_sigma;
+        varying float out_scale;
+        """
+
+        kde_dec = import_fury_shader(os.path.join("utils", f"{kernel.lower()}_distribution.glsl"))
+
+        kde_impl = """
+        float current_kde = kde(normalizedVertexMCVSOutput*out_scale, out_sigma);
+        color = vec3(current_kde);
+        fragOutput0 = vec4(color, 1.0);
+        """
+
+        kde_vs_dec = """
+        in float in_sigma;
+        varying float out_sigma;
+
+        in float in_scale;
+        varying float out_scale;
+        """
+
+
+        kde_vs_impl = """
+        out_sigma = in_sigma;
+        out_scale = in_scale;
+        """
+
+        tex_dec = import_fury_shader(os.path.join("effects", "color_mapping.glsl"))
+
+        tex_impl = """
+        // Turning screen coordinates to texture coordinates
+        vec2 res_factor = vec2(res.y/res.x, 1.0);
+        vec2 renorm_tex = res_factor*normalizedVertexMCVSOutput.xy*0.5 + 0.5;
+        float intensity = texture(screenTexture, renorm_tex).r;
+
+        if(intensity<=0.0){
+            discard;
+        }else{     
+            vec4 final_color = color_mapping(intensity, colormapTexture);
+            fragOutput0 = vec4(final_color.rgb, u_opacity*final_color.a);
+        }
+        """
+
+        fs_dec = compose_shader([varying_dec, kde_dec])
+
+        # Scales parameter will be defined by the empirical rule:
+        # 1*sima radius = 68.27% of data inside the curve
+        # 2*sigma radius = 95.45% of data inside the curve
+        # 3*sigma radius = 99.73% of data inside the curve
```

Comment:
"""
Scales parameter will be defined by the empirical rule:
1*sima radius = 68.27% of data inside the curve
 2*sigma radius = 95.45% of data inside the curve
3*sigma radius = 99.73% of data inside the curve
"""